app/src/main/java/com/fantastic/bookxchange/activities/AddBookActivity.java

```diff
@@ -1,13 +1,117 @@
 package com.fantastic.bookxchange.activities;
 
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
 import android.os.Bundle;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+
 import com.fantastic.bookxchange.R;
+import com.fantastic.bookxchange.models.Book;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Date;
 
-public class AddBookActivity extends BaseActivity  {
+public class AddBookActivity extends BaseActivity {
+
+    private static final String TAG = AddBookActivity.class.getSimpleName();
+    private EditText etBookAuthor;
+    private EditText etBookTitle;
+    private EditText etISBNNumber;
+    private EditText etDescription;
+    private Button btnAddBook;
+    private Button btnAddPhoto;
+    private ImageView ivPicture;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_add_book);
+        showToolbarBackButton();
+        enableToolbarBackButton();
+        etBookTitle = findViewById(R.id.etBookTitle);
+        etBookAuthor = findViewById(R.id.etBookAuthor);
+        etISBNNumber = findViewById(R.id.etISBNNumber);
+        etDescription = findViewById(R.id.etDescription);
+        btnAddBook = findViewById(R.id.btnAddBook);
+        btnAddPhoto = findViewById(R.id.btnAddPhoto);
+        ivPicture = findViewById(R.id.ivPicture);
+        ivPicture.setDrawingCacheEnabled(true);
+        ivPicture.buildDrawingCache();
+        btnAddPhoto.setOnClickListener(view -> {
+            Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+            startActivityForResult(intent, 0);
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        Bitmap bitmap = (Bitmap) data.getExtras().get("data");
+        ivPicture.setImageBitmap(bitmap);
+    }
+
+    public void addBook(View view) {
+        if (!validate(etBookTitle, etBookAuthor, etISBNNumber, etDescription)) {
+            return;
+        }
+        String isbn = getText(etISBNNumber);
+        uploadCoverPage(ivPicture.getDrawingCache(),isbn);
+
+    }
+    private void saveBook(String url){
```

Comment:
It's better to check if the book already exists on the openlibrary before making it from scratch, right? Also, probably we need to check if this book already exists on our DB.